common/authentication/postgresql/metadata_test.go

```diff
@@ -0,0 +1,100 @@
+package postgresql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildConnectionString(t *testing.T) {
+	tc := []struct {
+		name        string
+		metadata    PostgresAuthMetadata
+		expected    string
+		expectError bool
+	}{
+		{
+			name:        "empty metadata",
+			metadata:    PostgresAuthMetadata{},
+			expected:    "",
+			expectError: true,
+		},
+		{
+			name: "valid DSN connection string",
+			metadata: PostgresAuthMetadata{
+				ConnectionString: "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=my_db",
+			},
+			expected:    "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=my_db",
+			expectError: false,
+		},
```

Comment:
```suggestion
		{
			name: "valid DSN connection string",
			metadata: PostgresAuthMetadata{
				ConnectionString: "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=my_db",
			},
			expected:    "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=my_db",
			expectError: false,
		},
		{
			name: "valid DSN connection values",
			metadata: PostgresAuthMetadata{
				Host:           "localhost",
				User: 		"postgres",
				Password:       "example",
				Port:           "5432",
				Database:       "my_db",
			},
			expected:    "host=localhost user=postgres password=example port=5432 database=my_db",
			expectError: false,
		},
```

I would expect this test case should also pass

Reply:
with metadata only the order isn't preserved.. Added a specific test for this case :)